File: python/easy/1678.goalParser.py
class Solution:
    def interpret(self, command: str) -> str:
        r = command.replace("()", "o")
        res: str = ""

        for x in r:
            if x == "(" or x == ")":
                x = ""
            else:
                res += x

        return res


if __name__ == "__main__":
    solution = Solution()
    print(solution.interpret("G()(al)"))  # "Goal"
    print(solution.interpret("G()()()()(al)"))  # "Gooooal"

File: javascript/easy/1678.goalParser.js
/**
 * @param {string} command
 * @return {string}
 */
var interpret = function (command) {
    let r = command.replaceAll("()", "o")
    let str = ""

    for (let i = 0; i < r.length; i++) {
        if (r[i] == "(" || r[i] == ")") {
            r[i] = ""
        } else {
            str += r[i]
        }
    }

    return str
}

console.log(interpret("G()(al)")) // "Goal"
// Output: "Goal"
// Explanation: The Goal Parser interprets the command as follows:
// G -> G
// () -> o
// (al) -> al
// The final concatenated result is "Goal".

console.log(interpret("G()()()()(al)")) // "Gooooal"

console.log(interpret("(al)G(al)()()G"))
